storage/src/test/java/tech/pegasys/artemis/storage/MapDbDatabaseTest.java

```
@@ -80,6 +81,11 @@
 
         @Override
         public void onGenesis(Store store) {}
+
+        @Override
+        public SafeFuture<Optional<Store>> onStoreRequest() {
+          return null;
```

Should return a `failedFuture` instead of `null`.  While it's not called today getting a null future returned is a nasty surprise.

---

Could this extend `StubStorageUpdateChannel` and just override `onStorageUpdate`?

---

Yeah definitely, that's a great idea.

---

Done.